Signal/src/contact/SystemContactsFetcher.swift

```diff
@@ -434,15 +452,21 @@ class SystemContactsFetcher: NSObject {
 
         DispatchQueue.global().async {
 
-            var fetchedContacts: [Contact]?
+            guard self.tryToAcquireContactFetchLock() else {
+                Logger.info("\(self.TAG) ignoring redundant system contacts fetch.")
+                return
+            }
 
+            var fetchedContacts: [Contact]?
             switch self.contactStoreAdapter.fetchContacts() {
             case .success(let result):
                 fetchedContacts = result
             case .error(let error):
                 completion?(error)
+                self.releaseContactFetchLock()
                 return
             }
+            self.releaseContactFetchLock()
```

Comment:
maybe a good use for a swift `defer` block.

Reply:
Done.